engine.py

```diff
@@ -108,6 +116,57 @@ def post_app_init(self):
             except:
                 traceback.print_exc()
 
+    def validate_context(self, tank, context):
+        '''
+        Given the context provided at initialisation, if we're in a project context only, forces the user to select
+        a proper task context using rdokatana.taskChooser.taskChooser.TaskChooser.
+        :param tank: a Tank object instance.
+        :type tank: :class:`sgtk.tank.Tank`
+        :param context: the current context.
+        :type context: :class:`tank.context.Context`
+        :return: a proper working context.
+        :rtype: :class:`tank.context.Context`
+        '''
+        if context.project:
+            if not context.entity:
+                task = self.select_task_ui(context)
+                newContext = tank.context_from_entity('Task', task['id'])
+                return newContext
+            elif context.entity and not context.task:
+                filters = [
+                    ['project', 'is', context.project],
+                    ['entity', 'is', context.entity],
+                ]
+                task = None
+                if context.entity['type'] == 'Shot':
+                    filters.append(['content', 'is', 'Lighting'])
+                    task = tank.shotgun.find_one("Task", filters)
+                elif context.entity['type'] == 'Asset':
+                    filters.append(['content', 'is', 'Shading'])
+                    task = tank.shotgun.find_one("Task", filters)
+                if task:
+                    newContext = tank.context_from_entity('Task', task['id'])
+                    return newContext
+                else:
+                    task = self.select_task_ui(context)
+                    newContext = tank.context_from_entity('Task', task['id'])
+                    return newContext
```

Comment:
yeah I can probably refactor a bit.
There are 3 use cases though:
1/ No entity at all. We create the UI
2/ An entity but no task. We try to find a default task:
a) if found, all good
b) if not, we need to show the UI.